rskj-core/src/main/java/co/rsk/peg/StateForFederator.java

```diff
@@ -53,15 +52,13 @@ public SortedMap<Keccak256, BtcTransaction> getRskTxsWaitingForSignatures() {
      * @return The RLP-encoded byte array representing the current state.
      */
     public byte[] encodeToRlp() {
-        byte[] serializedRskTxsWaitingForSignatures = 
-            BridgeSerializationUtils.serializeRskTxsWaitingForSignatures(rskTxsWaitingForSignatures);
-        return RLP.encodeList(serializedRskTxsWaitingForSignatures);
+        return BridgeSerializationUtils.serializeRskTxsWaitingForSignatures(
+            rskTxsWaitingForSignatures);
     }
 
     private static byte[] decodeRlpToMap(byte[] rlpData) {
         Objects.requireNonNull(rlpData);
 
-        RLPList rlpList = (RLPList) RLP.decode2(rlpData).get(0);
-        return rlpList.get(0).getRLPData();
+        return RLP.decode2(rlpData).get(0).getRLPData();
```

Comment:
why this change?

Reply:
This only serializes the list twice creating a 2d serialized array. There is no apparent reason for doing this, maybe there is but not to my knowledge.

Reply:
i won't change anything serialization-related here, we could break consensus

Reply:
This is just that we were serializing the list twice, with the change we serialized once